packages/php-config-printer/src/NodeFactory/ConstantNodeFactory.php

```diff
@@ -28,6 +28,12 @@ public function createClassConstantIfValue(string $value, bool $checkExistence =
         $match = Strings::match($value, self::CLASS_CONST_FETCH_REGEX);
         if ($match !== null) {
             [$class, $constant] = explode('::', $value);
+
+            // Ignore static factories (FQCN::method)
+            if (method_exists($class, $constant)) {
+                return null;
+            }
```

Comment:
Neat trick :wink:

Reply:
Hah, I was wondering if you reject it since there was PHPStan rule disallowing `method_exists` usage 😅 But here it was the only way I knew to skip static factory usage without writing custom weird logic 🤷‍♂️

Reply:
In this context it is quite safe IMO. If the package is in same autoload scope as project, the method will be found and detected :slightly_smiling_face: 

Let's see :laughing: